src/main/java/com/palantir/gradle/revapi/RevapiPlugin.java

```diff
@@ -47,22 +49,37 @@ public void apply(Project project) {
 
         ConfigManager configManager = new ConfigManager(configFile(project));
 
-        TaskProvider<RevapiJavaTask> revapiTask = project.getTasks().register("revapi", RevapiJavaTask.class, task -> {
-            task.dependsOn(allJarTasksIncludingDependencies(project, revapiNewApi));
+        TaskProvider<RevapiReportTask> revapiTask = project.getTasks().register("revapi", RevapiReportTask.class,
+                task -> configureRevapiJavaTask(project, revapiNewApi, configManager, task));
 
-            task.configManager().set(configManager);
+        project.getTasks().findByName(LifecycleBasePlugin.CHECK_TASK_NAME).dependsOn(revapiTask);
 
-            task.newApiDependencyJars().set(revapiNewApi);
+        project.getTasks().register(VERSION_OVERRIDE_TASK_NAME, RevapiVersionOverrideTask.class, task -> {
+            task.configManager().set(configManager);
```

Comment:
you can configure all Revapi abstract tasks in one go since they all need this `configManager()`